File: auto-u2a-polit-backend/src/main/java/com/auto/u2a/polit/entity/Organization.java
package com.auto.u2a.polit.entity;

import jakarta.persistence.*;
import java.util.UUID;

@Entity
@Table(name = "organizations")
public class Organization extends BaseEntity {
    
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID id;
    
    @Column(name = "tenant_id")
    private String tenantId;
    
    @Column(name = "name")
    private String name;
    
    @Column(name = "code")
    private String code;
    
    @Column(name = "type")
    private String type;
    
    @Column(name = "status")
    private String status;
    
    @Column(name = "parent_id")
    private UUID parentId;
    
    @Column(name = "path")
    private String path;
    
    // Getters and setters
    public UUID getId() {
        return id;
    }
    
    public void setId(UUID id) {
        this.id = id;
    }
    
    public String getTenantId() {
        return tenantId;
    }
    
    public void setTenantId(String tenantId) {
        this.tenantId = tenantId;
    }
    
    public String getName() {
        return name;
    }
    
    public void setName(String name) {
        this.name = name;
    }
    
    public String getCode() {
        return code;
    }
    
    public void setCode(String code) {
        this.code = code;
    }
    
    public String getType() {
        return type;
    }
    
    public void setType(String type) {
        this.type = type;
    }
    
    public String getStatus() {
        return status;
    }
    
    public void setStatus(String status) {
        this.status = status;
    }
    
    public UUID getParentId() {
        return parentId;
    }
    
    public void setParentId(UUID parentId) {
        this.parentId = parentId;
    }
    
    public String getPath() {
        return path;
    }
    
    public void setPath(String path) {
        this.path = path;
    }
    
    // 内部枚举类
    public enum OrgStatus {
        ACTIVE, INACTIVE, DELETED
    }
    
    public enum OrgType {
        COMPANY, DEPARTMENT, TEAM, PROJECT
    }
}

File: auto-u2a-polit-frontend/src/types/organization.ts
/**
 * 组织类型
 */
export type OrganizationType = 'COMPANY' | 'DEPARTMENT' | 'TEAM' | 'GROUP'

/**
 * 组织状态
 */
export type OrganizationStatus = 'ACTIVE' | 'INACTIVE'

/**
 * 组织
 */
export interface Organization {
  /**
   * 组织ID
   */
  id: string

  /**
   * 组织名称
   */
  name: string

  /**
   * 组织编码
   */
  code: string

  /**
   * 组织类型
   */
  type: OrganizationType

  /**
   * 父组织ID
   */
  parentId?: string

  /**
   * 父组织名称
   */
  parentName?: string

  /**
   * 描述
   */
  description?: string

  /**
   * 状态
   */
  status: OrganizationStatus

  /**
   * 排序
   */
  sort: number

  /**
   * 创建时间
   */
  createTime: string

  /**
   * 更新时间
   */
  updateTime: string

  /**
   * 子组织
   */
  children?: Organization[]
}

/**
 * 组织创建请求
 */
export interface OrganizationCreateRequest {
  /**
   * 组织名称
   */
  name: string

  /**
   * 组织编码
   */
  code: string

  /**
   * 组织类型
   */
  type: OrganizationType

  /**
   * 父组织ID
   */
  parentId?: string

  /**
   * 描述
   */
  description?: string

  /**
   * 状态
   */
  status?: OrganizationStatus

  /**
   * 排序
   */
  sort?: number
}

/**
 * 组织更新请求
 */
export interface OrganizationUpdateRequest {
  /**
   * 组织名称
   */
  name?: string

  /**
   * 组织编码
   */
  code?: string

  /**
   * 组织类型
   */
  type?: OrganizationType

  /**
   * 父组织ID
   */
  parentId?: string

  /**
   * 描述
   */
  description?: string

  /**
   * 状态
   */
  status?: OrganizationStatus

  /**
   * 排序
   */
  sort?: number
}

/**
 * 组织查询参数
 */
export interface OrganizationQueryParams {
  /**
   * 组织名称
   */
  name?: string

  /**
   * 组织编码
   */
  code?: string

  /**
   * 组织类型
   */
  type?: OrganizationType

  /**
   * 父组织ID
   */
  parentId?: string

  /**
   * 状态
   */
  status?: OrganizationStatus

  /**
   * 页码
   */
  page?: number

  /**
   * 每页数量
   */
  size?: number
}
